Specify API version and kind of ServiceBinding CRD (#94)
k8s-service-bindings/spec: README.md

# Service Binding

> A Service Binding describes the connection between a [Provisioned Service](#provisioned-service) and an [Application Projection](#application-projection).  It is codified as a concrete resource type and the name of the kind **SHOULD** be `ServiceBinding`.  Multiple Service Bindings can refer to the same service.  Multiple Service Bindings can refer to the same application.  An exemplar CRD can be found [here][sb-crd].

Reviewer: "Should" is too weak, alternate values will break compatibility. Also adding the apiVersion in addition to kind.

```suggestion
A Service Binding describes the connection between a [Provisioned Service](#provisioned-service) and an [Application Projection](#application-projection).  It **MUST** be codified as a concrete resource type with API version `service.binding/v1alpha2` and kind `ServiceBinding`.  Multiple Service Bindings can refer to the same service.  Multiple Service Bindings can refer to the same application.  An exemplar CRD can be found [here][sb-crd].
```

Author: I have committed your changes and squashed commits!
